src/style.rs

```diff
@@ -237,8 +241,9 @@ impl ProgressStyle {
                     alt_style,
                 } => {
                     buf.clear();
-                    if let Some(formatter) = self.format_map.get(key.as_str()) {
-                        buf.push_str(&formatter(state).replace('\t', &" ".repeat(self.tab_width)));
+                    if let Some(tracker) = self.format_map.get(key.as_str()) {
+                        tracker.write(state, &mut buf);
+                        buf = buf.replace('\t', &" ".repeat(self.tab_width));
```

Comment:
I think we should do a `struct TabRewriter(&mut dyn fmt::Write);` with an `impl fmt::Write` that rewrites the tabs before writing them to the inner `fmt::Write` instead.

Does that seem manageable?

Reply:
Not sure I follow. Where would we pass tab_width to that?